### PR TITLE
feat(cluster): add support of the JSON type in cluster migration

### LINKED ISSUE #2583: slot migrate not support json type
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Version

v2.9.0

### Minimal reproduce step

1、start slot migrate with json type data

### What did you expect to see?

slot migrate work well on json data type

### What did you see instead?

not working

### Anything Else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by git-hulk**
This closes #2583.

### CODE REVIEW

File: src/types/redis_json.cc [hunk 1]
```diff
@@ -674,4 +674,12 @@ rocksdb::Status Json::Resp(engine::Context &ctx, const std::string &user_key, co
...
 }
 
+rocksdb::Status Json::FromString(const std::string &value, JsonValue *result) {
+  Slice rest = value;
+  JsonMetadata metadata;
+  auto s = ParseMetadata({kRedisJson}, &rest, &metadata);
+  if (!s.ok()) return s;
+  return parse(metadata, rest, result);
+}
+
```
Comment (PragmaTwice) [inline src/types/redis_json.cc]: It seems useless. `Json::read` has same function.

File: src/types/redis_json.h [hunk 2]
```diff
@@ -82,6 +82,7 @@ class Json : public Database {
 
   rocksdb::Status Resp(engine::Context &ctx, const std::string &user_key, const std::string &path,
                        std::vector<std::string> *results, RESP resp);
+  static rocksdb::Status FromString(const std::string &value, JsonValue *result);
```
Comment (PragmaTwice) [inline src/types/redis_json.h]: If it's really needed, we can rename it to `FromRawString`. `FromString` is a weird name here.

File: src/types/redis_json.cc (hunk 1 above)
Comment (PragmaTwice) [reply src/types/redis_json.cc]: Ahhh my fault. It's not same.

File: src/types/redis_json.h (hunk 2 above)
Comment (git-hulk) [reply src/types/redis_json.h]: Yes, `FromRawString` would be better.

File: src/types/redis_json.cc [hunk 3]
```diff
@@ -674,4 +674,12 @@ rocksdb::Status Json::Resp(engine::Context &ctx, const std::string &user_key, co
   return rocksdb::Status::OK();
 }
 
+rocksdb::Status Json::FromRawString(const std::string &value, JsonValue *result) {
```
Comment (PragmaTwice) [inline src/types/redis_json.cc]:
```suggestion
rocksdb::Status Json::FromRawString(std::string_view value, JsonValue *result) {
```

**Review by caipengbo — DISMISSED** [review]

**Review by PragmaTwice — APPROVED** [review]

**Pull request merged**